README.md

```diff
@@ -19,10 +20,49 @@ via autowiring.
 
 ## Features
 
-- Invoke callable or create an object of a given class.
-- Resolve dependencies by parameter types using the given [PSR-11](http://www.php-fig.org/psr/psr-11/) container.
-- Pass concrete dependency instances by type.
-- Pass arguments by name.
+ * Injects dependencies when calling functions and creating objects
+ * Resolves dependency objects by parameter [type declaration](https://www.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration) ("autowiring")
+ * Resolves dependencies by parameter name
+ * Fetches dependency objects from a PSR-11 compatible container
+ * Accepts additional custom arguments from an array
+ * Handles variadic parameters, i.e. `function (MyClass ...$a)`
+
+
+## About
+
+Injector can automatically resolve and inject dependencies when calling
+functions and creating objects.
+
+Its main purpose is to inject object arguments from a PSR-11 compatible *object
```

Comment:
```suggestion
Its main purpose is to inject object arguments from a PSR-11 compatible *service
```

Reply:
Hmm, what is a "service"? The container doesn't need to know about such a concept. All it really does is to return an object for a given class. The object can be anything.

Reply:
Both are, in fact, kind of wrong. PSR uses common term "dependency injection container". I think we should use it as well.